crates/autopilot/src/protocol/fee.rs

```diff
@@ -7,44 +7,79 @@
 use {
     crate::{
         arguments,
+        database::Postgres,
         driver_model::solve::{fee_policy_to_dto, FeePolicy},
     },
+    anyhow::{Context, Result},
     model::{
         auction::Auction,
         order::{OrderClass, OrderUid},
     },
-    std::collections::HashMap,
+    number::conversions::big_decimal_to_u256,
+    shared::order_validation::is_order_outside_market_price,
+    std::{collections::HashMap, sync::Arc},
 };
 
-pub struct Policies {
-    policies: HashMap<OrderUid, Vec<FeePolicy>>,
+pub struct PolicyFactory {
+    config: arguments::FeePolicy,
+    db: Arc<Postgres>,
 }
 
-impl Policies {
-    pub fn new(auction: &Auction, config: arguments::FeePolicy) -> Self {
-        Self {
-            policies: auction
+impl PolicyFactory {
```

Comment:
nit: I'm not sure this is a true [factory pattern](https://en.wikipedia.org/wiki/Factory_method_pattern), given the create method is called `build` (is this maybe more of a [builder pattern](https://en.wikipedia.org/wiki/Builder_pattern))?

I think this component should be made part of the auction building process and expose a single method that takes an auction without fees by value and returns an auction with fees applied.

Reply:
I did not use any specific pattern, I just aimed to group all data (that does not change over time) needed to build the fee policies in one object.

Reply:
I see, but by using nomenclature of such patterns future maintainers may think that a specific pattern was used/intended here. Can we call it FeeEngine or something less loaded in this case?

Also, this could be the first component of our autopilot domain (db should probably be an infra type that doesn't leak the fact that Postgres is used).